honeybadger/contrib/celery.py

```diff
@@ -0,0 +1,94 @@
+from celery import current_task
+from celery.signals import task_failure, task_postrun
+from honeybadger import honeybadger
+from honeybadger.plugins import Plugin, default_plugin_manager
+
+
+class CeleryPlugin(Plugin):
+    def __init__(self):
+        super().__init__("Celery")
+
+    def supports(self, config, context):
+        """
+        Check whether this is a a celery task or not.
+        :param config: honeybadger configuration.
+        :param context: current honeybadger configuration.
+        :return: True if this is a celery task, False else.
+        """
+        return current_task != None
+
+    def generate_payload(self, default_payload, config, context):
+        """
+        Generate payload by checking celery task object.
+        :param context: current context.
+        :param config: honeybadger configuration.
+        :return: a dict with the generated payload.
+        """
+        payload = {
+            "component": current_task.__module__,
+            "action": current_task.name,
+            "params": {
+                "args": list(current_task.request.args),
+                "kwargs": current_task.request.kwargs,
+            },
+            "context": {
+                "task_id": current_task.request.id,
+                "retries": current_task.request.retries,
+                "max_retries": current_task.max_retries,
+            },
```

Comment:
not sure if that should be in the context, maybe a new `task_info` section?

Reply:
I think context is the right section to add it. All payload follows the schema defined [here](https://github.com/honeybadger-io/honeybadger-schemas/blob/77bcff997132d5393b51db26d1cc77efd6f0dc88/exceptions/v1/notices.json#L184)